app/index.tsx

```diff
@@ -150,103 +137,107 @@ export default class Root extends React.Component<{}, IState> {
 
 		// Open app from app icon
 		store.dispatch(appInit());
-	};
+	}, []);
 
-	getMasterDetail = (width: number) => {
+	const getMasterDetail = (width: number) => {
 		if (!isTablet) {
 			return false;
 		}
 		return width > MIN_WIDTH_MASTER_DETAIL_LAYOUT;
 	};
 
-	setMasterDetail = (width: number) => {
-		const isMasterDetail = this.getMasterDetail(width);
+	const setMasterDetail = (width: number) => {
+		const isMasterDetail = getMasterDetail(width);
 		store.dispatch(setMasterDetailAction(isMasterDetail));
 	};
 
 	// Dimensions update fires twice
-	onDimensionsChange = debounce(({ window: { width, height, scale, fontScale } }: { window: IDimensions }) => {
-		this.setDimensions({
-			width,
-			height,
-			scale,
-			fontScale
-		});
-		this.setMasterDetail(width);
-	});
-
-	setTheme = (newTheme = {}) => {
-		// change theme state
-		this.setState(
-			prevState => newThemeState(prevState, newTheme as IThemePreference),
-			() => {
-				const { themePreferences } = this.state;
-				// subscribe to Appearance changes
-				subscribeTheme(themePreferences, this.setTheme);
-			}
+	const onDimensionsChange = useCallback(
+		debounce(({ window: { width, height, scale, fontScale } }: { window: IDimensions }) => {
+			// We need this batchedUpdates because we are calling this from a debounce and
+			// the React isn't able to batch multiples setState inside a setTimeout
+			// https://stackoverflow.com/a/48610973
+			unstable_batchedUpdates(() => {
+				setDimensions({
+					width,
+					height,
+					scale,
+					fontScale
+				});
+				setMasterDetail(width);
+			});
+		}),
+		[]
+	);
+
+	const setNewTheme = (newThemePreference = {}) => {
+		const { theme: newTheme, themePreferences: newThemePreferences } = newThemeState(
+			themePreferences,
+			newThemePreference as IThemePreference
```

Comment:
You're kind of just making TS happy here.
You should avoid `as` as much as you can.

```tsx
const setNewTheme = (newThemePreference: IThemePreference) => {
	const { theme: newTheme, themePreferences: newThemePreferences } = newThemeState(
		themePreferences,
		newThemePreference
	);
	setThemePreferences(newThemePreferences);
	setTheme(newTheme);
	subscribeTheme(themePreferences, setNewTheme);
};

...

subscribeTheme = (themePreferences: IThemePreference, setTheme: Function): void
```

Please evaluate the need of `{}` there.
I don't think it's needed.

Reply:
you are right!